### PR TITLE
fix: allow array of strings for `output.library.root`

### LINKED ISSUE #7176: `output.library.root` should accept an array
**Do you want to request a *feature* or report a *bug*?** 

Bug?

**What is the current behavior?**

In `webpack.conf.js`:

```javascript
export = {
    output: {
        libraryTarget: "umd",
        library: {
            amd: "my-library",
            commonjs: "my-library",
            root: ["My", "Library"],
        }
    }
};
```

Result:

```
Invalid configuration object. Webpack has been initialised using a configuration object that does not match the API schema.
 - configuration.output.library should be one of these:
   string | [string] | object { root?, amd?, commonjs? }
   -> If set, export the bundle as library. `output.library` is the name.
   Details:
    * configuration.output.library should be a string.
    * configuration.output.library should be an array:
      [string]
    * configuration.output.library.root should be a string.
      -> Name of the property exposed globally by a UMD library
```

**If the current behavior is a bug, please provide the steps to reproduce.**

Create the configuration above and run `webpack`.

Note that I can pass an array directly to the library as shown below. In that case, the correct object is added to `root`, but `amd` and `commonjs` are registered with the wrong name.

```javascript
export = {
    output: {
        libraryTarget: "umd",
        library: ["My", "Library"],
    }
};
```

So why can I pass an array to `library`, but not to `library.root`?

**What is the expected behavior?**

No error.

**If this is a feature request, what is motivation or use case for changing the behavior?**

**Please mention other relevant information such as the browser version, Node.js version, webpack version, and Operating System.**

- webpack: 4.6.0
- node: 8.9.4
- OS: Windows 8.1

### PR DESCRIPTION
**Pull request by byzyk**
**What kind of change does this PR introduce?**

bugfix

**Did you add tests for your changes?**

configCases

**If relevant, link to documentation update:**

N/A

**Summary**

Closes #7176

**Does this PR introduce a breaking change?**

No

### CODE REVIEW

**Comment by byzyk** [comment]
I assume tests will be needed for this one

**Comment by montogeek** [comment]
Isn't needed a change in the source code?

**Comment by byzyk** [comment]
@montogeek quite possible. Need to add a test case first and will see, probably tom

File: test/configCases/library/umd/webpack.config.js [hunk 1]
```diff
@@ -1,10 +1,22 @@
-module.exports = {
-	output: {
-		libraryTarget: "umd",
-		library: {
-			root: "testLibrary",
-			amd: "test-library",
-			commonjs: "test-library"
+module.exports = [
```
Comment (montogeek) [inline test/configCases/library/umd/webpack.config.js]: I am not sure if it is fine to edit an existing test.

Comment (byzyk) [reply test/configCases/library/umd/webpack.config.js]: They are quite similar so I thought they can live in the same folder. But it's not a big deal I can create a separate case. Interesting that I did exactly that in the first place but then I thought why not together? 😃

**Comment by byzyk** [comment]
Added a test case. Seems it compiles without errors

**Comment by byzyk** [comment]
I checked with the issue author and he confirmed it works as expected. Is there anything else pending to get this one merged?

**Comment by sokra** [comment]
Thanks

**Pull request merged**